vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt

```diff
@@ -95,12 +159,45 @@ class UserListViewModel @AssistedInject constructor(@Assisted initialState: User
     private fun handleClearSearchUsers() {
         knownUsersSearch.accept("")
         directoryUsersSearch.accept("")
+        identityServerUsersSearch.accept("")
         setState {
             copy(searchTerm = "")
         }
     }
 
     private fun observeUsers() = withState { state ->
+
+        identityServerUsersSearch
+                .filter { it.isEmail() }
+                .throttleLast(300, TimeUnit.MILLISECONDS)
+                .switchMapSingle { search ->
+                    val rx = session.rx()
+                    val stream =
+                            rx.lookupThreePid(ThreePid.Email(search)).flatMap {
+                                it.getOrNull()?.let { foundThreePid ->
+                                    rx.getProfileInfo(foundThreePid.matrixId)
+                                            .map { json ->
+                                                ThreePidUser(
+                                                        email = search,
+                                                        user = User(
+                                                                userId = foundThreePid.matrixId,
+                                                                displayName = json[ProfileService.DISPLAY_NAME_KEY] as? String,
+                                                                avatarUrl = json[ProfileService.AVATAR_URL_KEY] as? String
+                                                        )
+                                                )
+                                            }
+                                            .onErrorResumeNext {
+                                                Single.just(ThreePidUser(email = search, user = User(foundThreePid.matrixId)))
+                                            }
+                                } ?: Single.just(ThreePidUser(email = search, user = null))
+                            }
+                    stream.toAsync {
+                        copy(matchingEmail = it)
+                    }
+                }
+                .subscribe()
+                .disposeOnClear()
```

Comment:
...and replace this two lines with:
```kotlin
                .execute { it ->
                    copy(matchingEmail = it)
                }
```
(Not sure this is equivalent though)

Reply:
seen with granfra, it's not

Reply:
OK